### PR TITLE
Add ready replicas to deployment config

### PR DESCRIPTION
**Pull request by mfojtik**
Implements [this](https://github.com/kubernetes/kubernetes/pull/37959) upstream feature downstream.

Fixes: https://github.com/openshift/origin/issues/12415

### CODE REVIEW

**Comment by mfojtik** [comment]
[test]

File: pkg/deploy/util/util.go [hunk 1]
```diff
@@ -375,6 +375,16 @@ func GetAvailablePods(pods []*api.Pod, minReadySeconds int32) int32 {
 	return available
 }
 
+// GetReadyReplicaCountForReplicaSets returns the number of ready pods corresponding to
+// the given replication controller.
+func GetReadyReplicaCountForReplicaSets(replicationControllers []api.ReplicationController) int32 {
```
Comment (mfojtik) [inline pkg/deploy/util/util.go]: hups

**Comment by mfojtik** [comment]
post-rebase fixup done.

**Comment by kargakis** [comment]
lgtm [merge]

**Comment by mfojtik** [comment]
added openapi... [merge]

**Comment by mfojtik** [comment]
flake: https://github.com/openshift/origin/issues/12362

[merge]

**Pull request merged**